storage/remote/queue_manager.go

```diff
@@ -486,6 +487,46 @@ outer:
 	return true
 }
 
+func (t *QueueManager) AppendExemplars(exemplars []record.RefExemplar) bool {
+outer:
+	for _, e := range exemplars {
+		t.seriesMtx.Lock()
+		lbls, ok := t.seriesLabels[e.Ref]
+		if !ok {
+			// todo: metric to track dropped exemplars?
```

Comment:
I think a metric would be a good idea as this case could be quite mysterious otherwise.

Reply:
A metric to track dropped exemplars is easy enough, since this happens before we get to an individual shard.

However if we want to track exemplarsTotal, exemplarsFailedTotal, etc. That gets a bit more complex. I think the right solution would be to have a separate channel for exemplars per shard, count pending samples and pending exemplars separately, and have `buildWriteRequest` take two `TimeSeries` slices (samples and exemplars, just append the exemplars slice to the samples slice when building the actual protobuf for sending).

This would allow us to properly count samples vs exemplars sent total, failed, and dropped/failed in the shutdown case where we include the samples in the queue.

I can make that change easily enough, but lets agree whether we want to do it, or if just `droppedExemplars` is sufficient for now.

Reply:
Interesting point you bring up, I think I am fine with just the dropped exemplars, though you could add a total exemplars in this function as well. We can do the larger change later if it is needed.

Reply:
We could add `exemplarsTotal` here as well, but I was just looking at where the rest of the metrics are updated and it's all in shards functions afaict, so I wanted the other exemplar metrics to be updated there as well. I don't have a strong opinion as to when we add anything other than `dropped` for exemplars.